dev/app/components/main/section-form/dialogs/add-projects/add-projects.controller.js

```diff
@@ -0,0 +1,92 @@
+AddProjectsController.$inject = ['$scope', 'close', 'TbUtils', 'projects', 'filterFilter'];
+
+function AddProjectsController($scope, _close, TbUtils, projects, filterFilter) {
+    var vm = this;
+
+    vm.options = {};
+    vm.options.startingPage = 0;
+    vm.options.pageSize = 60;
+    vm.options.count = 0;
+    vm.limitInLettersToSearch = 3;
+    vm.projects = [];
+    vm.totalProjects = [];
+    vm.projectsPagination = [];
+    vm.projectsSelected = projects.selectedProjectsInSectionForm;
+    vm.onPageChange = onPageChange;
+    vm.addProject = addProject;
+    vm.showSelectedProjects = showSelectedProjects;
+
+    $scope.accept = accept;
+    $scope.close = close;
+    projects.getProjectsCount(getProjectsCountSuccess);
+
+    function getProjectsCountSuccess(response){
+        vm.options.count = response.data;
+        projects.getProjectsWithPagination(vm.options.startingPage, vm.options.pageSize , getProjectsSuccess, getProjectsFail);
+    }
+
+    function getProjectsSuccess(response) {
+        TbUtils.fillListWithResponseData(response.data, vm.projectsPagination);
+        vm.projectsLoading = false;
+    }
+    
+    function getProjectsFail(response) {
+        TbUtils.showErrorMessage('error', response,
+                                 'No se ha podido obtener los proyectos deseados.',
+                                 'Error');
+
+        vm.projectsLoading = false;
+    }
+
+    function onPageChange(skip, page) {
+        if($scope.search) $scope.search.data = '';
+        vm.projects = vm.projectsPagination;
+        vm.projects.length = 0;
+        vm.projectsLoading = true;
+        projects.getProjectsWithPagination( page, skip, getProjectsSuccess, getProjectsFail);
+    }
+
+    projects.getProjects(getTotalProjectsSuccess, getTotalProjectsFail);
+
+    function getTotalProjectsSuccess(response) {
+        console.log(response);
+        TbUtils.fillListWithResponseData(response.data, vm.totalProjects);
+    }
+
+    $scope.$watch('search.data', function(term) {
+        let obj = {Name: term};
+
+        if(term && term.length >= vm.limitInLettersToSearch) {
+            $scope.filterProjects = filterFilter(vm.totalProjects, obj);
+            vm.projects = $scope.filterProjects;
+        }
+
+        else {
+            console.log('vacio');
+            vm.projects = vm.projectsPagination;
+        }
+    });
```

Comment:
@danielzy95 crees que podes pasar esto al directivo de search? Es que veo que se hace lo mismo en secciones e inclusive yo hice un copy and paste en finiquitos porque no está implementado en el directivo y veo que estamos repitiendo.
`    $scope.$watch('search.data', function(term) {
         let obj = {Name: term};
 
         if(term && term.length >= vm.limitInLettersToSearch) {
             $scope.filterProjects = filterFilter(vm.totalProjects, obj);
             vm.projects = $scope.filterProjects;
         }
 
         else {
             console.log('vacio');
             vm.projects = vm.projectsPagination;
         }
    });`